### ISSUE TITLE
Accessibility: ipython.org

### ISSUE THREAD

**Issue by Joel-Ando** [jupyterhub/outreachy#82]
### What page is this for?

https://ipython.org/index.html#

### WAVE accessibility report

![ipython](https://user-images.githubusercontent.com/81184842/196422621-020e67ae-1d0b-452f-90c6-8f469f74833d.JPG)
1 LANGUAGE MISSING OR UNIDENTIFIED: this indicate a missing <HTML> language attribute, which is a necessity for all web pages that allows users know which language they are interacting with on the page, search engines and screen readers
solution adding a proper <html lang = "en"> attribute at the <head> of the webpage
2 CONTRAST ERRORS: There are low contrast between texts and background. Users especially those with low vision will find it difficult to distinguish text from background.
Solution: increase the brightness between the texts and the background.
3 ALERTS: There are a good number of alerts.
-Fully justified text is present: block of text that stretch with justification can negatively impact readability due to white spacing that exist between word.
solution: remove the full justification completely and also play around with letter-spacing or left-alignment.
-Suspicious Alternative text: Alternative text is either insufficient or is unrelated to the IPYTHON image hence content would not be available if image is not available to screen-readers.
solution : add alt text attribute which describes the image <alt ="ipython interactive computing logo">
-Suspicious link text: Links, which are often read out of context, should clearly describe the destination of the link. text like ( click here) are confusing to the users because it is not clear enough to describe its purpose.
solution: instead of "click Here" "IPython’s Notebook Viewer" should be use as the link text.
-image no alt value: complex images( pie chart, bar chart, maps etc) like the IPYTHON NOTE BOOK VIEWER should have alt text that describes the image as this type of images become difficult to understand
solution: add alt text with proper description to image when possible.
4 STRUCTURAL ELEMENTS: We also have some structure elements like <h1> <h3> <h4>. HTML heading tags should follow the right hierarchy h1, h2, h3, h4 . and properly structure on web page
-Unordered(bulleted list) is present: ensure that if using an <ul> on a webpage it is appropriate for the context. On the IPYTHON web page the ANNOUNCEMENT section should use ORDERED list since the content is sequential
-Inline frame <iframe> is present: iframes are read as if it were part of the page that contains it that is it should be easily
accessed.
Solution: the iframe at the footer should be made visible and not hidden to ease accessibility.
Solution the aria-label labels the <nav> as main navigation which will be confusing with the <main> content to users
aria-label should label nav with <aria-label="nav-menu"> for better accessibility.

### The HTML element to be changed

```html
<!--visibility of iframe hidden at the footer-->
<!-- either it is remove completely or made visible-->
<iframe scrolling="no" frameborder="0" allowtransparency="true" src="https://platform.twitter.com/widgets/widget_iframe.7dae38096d06923d683a2a807172322a.html?origin=https%3A%2F%2Fipython.org" title="Twitter settings iframe" style="display: none;"></iframe>
Inline frame
<iframe id="rufous-sandbox" scrolling="no" frameborder="0" allowtransparency="true" allowfullscreen="true" 
style="position: absolute; 
      visibility: hidden; // visibility of iframe is hidden and hence cannot be access
         display: none; 
           width: 0px; 
          height: 0px; 
         padding: 0px; 
          border: none;
" title="Twitter analytics iframe">
</iframe>
```

### Your proposed HTML (or CSS), after change

```html
<!--iframe css style in footer made visible-->

<iframe scrolling="no" frameborder="0" allowtransparency="true" src="https://platform.twitter.com/widgets/widget_iframe.7dae38096d06923d683a2a807172322a.html?origin=https%3A%2F%2Fipython.org" title="Twitter settings iframe" 
style="display: none;">
</iframe>
Inline frame
<iframe id="rufous-sandbox" scrolling="no" frameborder="0" allowtransparency="true" allowfullscreen="true" 
style="position: absolute; 
      visibility: visible; //iframe visibility made visible hence can be access
         display: none; 
           width: 0px; 
          height: 0px; 
         padding: 0px; 
          border: none;
" title="Twitter analytics iframe">
</iframe>
```

### More info

I don't know if the Accessibility of jupyterhub web pages considers the devices or the screens sizes on which these pages will be accessed.?
If not then its an issue to be address, because a page can be accessible on the a desktop or laptop screen web browser but not on a handheld or mobile device web browser like mobile phones and tablets.
Now our days handheld or mobile devices have become more popular and sophisticated so much that they can be use to perform complex computational operations. many are beginning to use this devices to brows and surf the internet, in scientific and educational laboratories to carry out analytical operations.
![ipython1](https://user-images.githubusercontent.com/81184842/196419036-4a4c9d7f-7434-4c46-8c4e-a7084e925da2.JPG)
![Capture1](https://user-images.githubusercontent.com/81184842/196419120-e5b3c81a-80d5-420f-abd3-0237e9ea03cf.JPG)

from the screenshots above i used **chrome dev tools** to **evaluate accessibility in terms of responsiveness of IPYTHON webpage.**
I would suggest that the responsiveness of this page be tackled so that the ease with which this page is accessed with wide screens should be the same on a smaller screen.
We can see that there is a lot information on the smaller screen of the **ipython** making it difficult for people with disabilities especially with low vision to access content using this devices. for proper responsiveness the side menu should go underneath the main page. this way it reduces 
I also suggest the **Announcement section** and its content be remove and only popup as an UPDATE since the same content  can be access in the **news menu**  and avoid unnecessary repetition.

**Comment by minrk** [comment]
I think you've found the hidden twitter iframe which is meant to be hidden. I think it should probably be removed!

Thanks for the other suggestions as well!

**Comment by Joel-Ando** [comment]
> I think you've found the hidden twitter iframe which is meant to be hidden. I think it should probably be removed!
> 
> Thanks for the other suggestions as well!

thanks Mr @minrk

**Comment by sgibson91** [comment]
Thank you for your contribution! Since the contribution period is now over, we will close this issue.
